ark/mibi/tiling_utils.py

```diff
@@ -0,0 +1,240 @@
+        print("Error: moly point run parameter must be either 0 or 1")
+        moly_run_insert = int(
+            input("Insert moly points between runs? Enter 0 for no and 1 for yes: ")
+        )
+
+    tiling_params['moly_run'] = moly_run_insert
+
+    # whether to insert moly points between tiles
+    # NOTE: moly points will be inserted between different runs regardless of what's set here
+    moly_interval_insert = int(
+        input("Specify moly point tile interval? Enter 0 for no and 1 for yes: ")
+    )
+
+    while moly_interval_insert not in [0, 1]:
+        print("Error: moly interval insertion parameter must enter 0 or 1")
+        moly_interval_insert = int(
+            input("Specify moly point tile interval? Enter 0 for no and 1 for yes: ")
+        )
+
+    # if moly insert is set, we need to specify an additional moly_interval param
+    # NOTE: the interval applies regardless of if the tiles overlap runs or not
+    if moly_interval_insert:
+        moly_interval = int(input("Enter the fov interval size to insert moly points: "))
+
+        while moly_interval < 1:
+            print("Error: moly interval must be positive")
+            moly_interval = int(input("Enter the fov interval size to insert moly points: "))
+
+        tiling_params['moly_interval'] = moly_interval
+
+    return tiling_params, moly_point
+
+
+def create_tiled_regions(tiling_params, moly_point):
+    """Create the tiled regions for each fov
+
+    Args:
+        tiling_params (dict):
+            The tiling parameters created by set_tiling_params
+        moly_point (dict):
+            The moly point to insert between fovs (and intervals if specified in tiling_params)
+
+    Returns:
+        dict:
+            Data containing information about each tile, will be saved to JSON
+    """
+
+    # helper function for creating all pairs between two lists
+    def pairs(*lists):
+        for t in combinations(lists, 2):
+            for pair in product(*t):
+                yield pair
+
+    # get the current time info
+    dt = datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S')
+
+    # define the fov tiling info
+    tiled_regions = {
+        'exportDateTime': dt,
+        'fovFormatVersion': tiling_params['fovFormatVersion'],
+        'fovs': []
+    }
+
+    # define a counter to determine where to insert a moly point
+    # only used if tiling_params['moly_interval'] is set
+    moly_counter = 0
+
+    # iterate through each region and append created tiles to tiled_regions['fovs']
+    for region_index in range(len(tiling_params['fov_num_x'])):
+        # extract start coordinates
+        start_x = tiling_params['region_start_x'][region_index]
+        start_y = tiling_params['region_start_y'][region_index]
+
+        # generate range of x and y coordinates
+        x_range = list(range(tiling_params['fov_num_x'][region_index]))
+        y_range = list(range(tiling_params['fov_num_y'][region_index]))
+
```

Comment:
Noticing a pattern where most tiling params need to be accessed as `tiling_params['some_parameter'][region_index]`.  If the structure of `tiling_params` is fixed that's fine, but would there be any downside changing the `tiling_params` output of `set_tiling_params` to be structured like:

```
{
    ...,
    fovs: ...,
    region_params: [{
        region_start_x: 0,
        region_start_y: 0,
        ...,
    }, ...],
    ...,
}
```
instead of:
```
{
    ...,
    fovs: ...,
    region_start_x: [...],
    region_start_y: [...],
    ...,
}
```
for the syntax simplification:
```
for region_params in tiling_params['region_params']:
    start_x = region_params['region_start_x']
    ...
```